Automattic/jetpack
# Boost: Update Minify UI

## Proposed changes:

* Split minify into two separate optimization modules (that still share some [common logic](https://github.com/Automattic/jetpack/blob/3234245d63e350335143175f1a4bce451058ddf6/projects/plugins/boost/app/modules/optimizations/minify/Minify.php)) - JS concatenation and CSS concatenation;
* Update UI accordingly and add options for exclude lists
* Update code to use data from new options;
* Remove some unnecessary checks left over from page optimize (like checking if [concatenation is enabled while it's enabled](https://github.com/Automattic/jetpack/pull/30005/files#diff-bdbd0ed4cf5358b0d543650616e2b11fe387d218debc97be8a5725813351ecd5L58))

### Other information:


## Jetpack product discussion



p1HpG7-lAA-p2

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






* Setup this version of Boost on your website;
* Enable CSS/JS concatenation;
* Check the front-end to make sure files are concatenated;
* Add/remove some items from the exclusion lists and check if they are ignored/not ignored.

## Notes:
* I've removed the GET params used to test [CSS](https://github.com/Automattic/jetpack/pull/30005/files#diff-924c0132da480bb9231bf48da1991027597c8ce38000fb7e4849944fb70d0258L103)/[JS](https://github.com/Automattic/jetpack/pull/30005/files#diff-924c0132da480bb9231bf48da1991027597c8ce38000fb7e4849944fb70d0258L77) concatenation. I did that because this isn't something we do in Boost. We only have a GET param (`jb-disable-modules`) for disabling modules.
* I've also removed page optimize's ["Non-critical script execution mode (experimental)" functionality](https://github.com/Automattic/jetpack/pull/30005/files#diff-924c0132da480bb9231bf48da1991027597c8ce38000fb7e4849944fb70d0258L158), since we already handle that in Boost under the `Defer Non-Essential JavaScript` optimization.
